### ISSUE TITLE
ghc make documentation is missing required libraries

### ISSUE THREAD

**Issue by peschkaj** [habit-lang/alb#5]
Installation using `ghc make` will fail unless specific libraries are installed via cabal. These should be documented, or documentation could be changed to insist that `cabal install` be used first and then `ghc make` can be used for every build after the first.

**Comment by fxdpntthm** [comment]
Do you mean:
```
cabal sandbox init
cabal configure
cabal install --dependencies-only
cabal install alb
```

**Comment by peschkaj** [comment]
Nope, that's not what I meant. `README` identifies GHC make as a method of installing `alb`, but the command provided will fail unless the dependencies have previously been met.

Of the four installation mechanisms listed in `README`, the one line `cabal install` mechanism is the only mechanism which works on a fresh system. Ideally, the stack installation mechanism would work, but it's currently broken. I opened an issue for installing via stack (see #4), but that was immediately closed.

**Comment by fxdpntthm** [comment]
```
cabal configure
cabal install --dependencies-only
```
Will install the needed dependencies. You can then run `ghc make` to build and generate alb executable. Using `ghc make` on a fresh install may not work as you many not have all the libraries needed for compilation. `cabal install` will also fail on a fresh system for the same reason. The `README` states how to generate the alb executable, it does not specify how you obtain the required dependencies in your workspace.  From `README`:
> DEPENDENCIES
> 
> Albatross attempts to be relatively self-contained; however, it does require some packages that may
> or may not be installed on your machine.  The alb.cabal file should have up-to-date dependencies.

`cabal install --dependencies-only` is one way of doing it, you may do it manually as well. After running the 2 cabal commands mentioned above, `ghc make` will work. 

If `cabal install` worked in first go, it implies, you have the needed
dependencies in your workspace, but `ghc make` cannot access it. You may use `cabal exec bash` (change `bash` to your favorite shell) to set a correct build path and then run `ghc make` to succeed.

>  Ideally, the stack installation mechanism would work, but it's currently broken. I opened an issue for installing via stack (see #4), but that was immediately closed.

I don't use stack so I don't know

**Comment by peschkaj** [comment]
Alright, I'll push a documentation update so that this isn't lost in a closed issue.

**Comment by peschkaj** [comment]
closed via 9ff557a212c35288a4f20531f88e3a73b002ce6a
